scripts/dts/edtlib.py

```diff
@@ -346,19 +346,17 @@ def _init_compat2binding(self):
                 # file that happened to match a compatible).
                 continue
 
-            # Do not allow two different bindings to have the same
-            # 'compatible:'/'on-bus:' combo
-            old_binding = self._compat2binding.get((binding.compatible,
-                                                    binding.on_bus))
-            if old_binding:
-                msg = (f"both {old_binding.path} and {binding_path} have "
-                       f"'compatible: {binding.compatible}'")
-                if binding.on_bus is not None:
-                    msg += f" and 'on-bus: {binding.on_bus}'"
-                _err(msg)
-
-            # Register the binding.
-            self._compat2binding[binding.compatible, binding.on_bus] = binding
+            # Register the binding in self._compat2binding, along with
+            # any child bindings that have their own compatibles.
+            self._register_binding(binding)
+            compat = binding.compatible
+            binding = binding.child_binding
+            while binding is not None:
+                if binding.compatible != compat:
+                    # Child bindings inherit their parents' compatibles if they
```

Comment:
I agree. This solution is a hack which preserves the workaround we added in 0d4dca10b2ac156b598f96b4dadb44ac7001380b for twister's sake. I wrote it this way because I think reverting that behavior is not acceptable for 2.5; this is a bit more duct tape which is trying to handle both issues.